nntrainer/layers/layer_node.cpp

```diff
@@ -223,9 +223,10 @@ void LayerNode::setOutputConnection(unsigned nth, const std::string &name,
   }
 
   auto &con = output_connections[nth];
-  NNTR_THROW_IF(con, std::invalid_argument)
-    << "cannot override connection, this slot is reserved for "
-    << con->toString();
+  // Should be override connection for the batch normalization realizer
+  // NNTR_THROW_IF(con, std::invalid_argument)
```

Comment:
Why don't we make a warning or make a another function such as resetOutputConnection?

Reply:
there are two situation we call the realizer from now on. one is for the training  and one for the inference. This kind of waring or error is the case for the training, not for the inference. For bn normalization case, the input of this bn realizer is graph representation which is already compiled and we need the modify or override the connection due to removed bn layer. In this case, it is not the error.

Reply:
no. it is not. There might be another cases like bn realize which modifies connections.